torch/_subclasses/fake_tensor.py

```diff
@@ -357,6 +359,22 @@ def from_real_tensor(
         # not yet supported in metatensors
         if t.is_quantized:
             raise UnsupportedFakeTensorException("quantized nyi in meta tensors")
+
+        # TODO: a very dirty shortcut for sparse.
+        if t.layout in torch.SPARSE_LAYOUTS:
+            with no_dispatch():
+                meta = TensorMetadata(t.dtype, t.shape, t.stride, t.device, t.layout,
+                                      None, t.storage_offset, t.requires_grad,
+                                      t.is_quantized, t.is_conj, t.is_neg, t.is_inference,
+                                      t.is_sparse, t.is_coalesced, t.dense_dim, t.sparse_dim)
+                meta_t = torch.tensor(t.shape, dtype=t.dtype,
+                                      requires_grad=t.requires_grad, device="meta",
+                )
+                return FakeTensor(fake_mode, meta_t, existing_device,
+                                 constant=t if make_constant else None,
+                                 sparse_layout=t.layout
+                )
```

Comment:
Could we avoid introducing this block and fix layout-specific issues (such as the lack of strides) at the lower levels (where strides are accessed)?

Reply:
Yeah, I started with that , and then just decided to add this very ugly intercept at the top (with the "a very dirty shortcut for sparse"). Let me try to dig a bit deeper into the lower levels.